### DISCUSSION TITLE
Is it expected that topics are set to ""?

### DISCUSSION THREAD

**Discussion by emiltin** [emqx/emqx#12929] [General]
Hi,
I'm using the emqx broker in a docker container, without just the default config.

When a client publish to a topic with Topic-Alias-Maximum set, emqx set an alias, so this is what a client subscribing to the topic receives:

```elixir
%{
  dup: false,
  via: #Port<0.11>,
  payload: "21",
  properties: %{"Topic-Alias": 1},
  retain: false,
  qos: 1,
  topic: "temp",
  packet_id: 1,
  client_pid: #PID<0.205.0>
}
```

When publishing again to the same topic, it set topic to "":

```elixir
%{
  dup: false,
  via: #Port<0.11>,
  payload: "21",
  properties: %{"Topic-Alias": 1},
  retain: false,
  qos: 1,
  topic: "",
  packet_id: 2,
  client_pid: #PID<0.205.0>
}
```

Is this expected that the topic is ""? 

See also this issue in the emqtt repo: https://github.com/emqx/emqtt/issues/240

**Comment by zmstone** [comment]
yes,
this is according to the spec.
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901113
quote

> If a Topic Alias mapping has been set at the receiver, a sender can send a PUBLISH packet that contains that Topic Alias and **a zero length Topic Name**. The receiver then treats the incoming PUBLISH as if it had contained the Topic Name of the Topic Alias.

**Reply by emiltin** [reply]
OK that's clear, thank you for the answer.

**Comment by emiltin** [comment]
"sender can send a PUBLISH" - does that refer to the broker, or some client?

**Reply by zmstone** [reply]
In MQTT. PUBLISH message is mutual: client can send to broker, broker can send to client.
